modules/social_features/social_private_message/social_private_message.install

```diff
@@ -359,21 +359,115 @@ function social_private_message_update_11005(): void {
  * Delete the default message template provided by private_message.
  */
 function social_private_message_update_11006(): void {
-  // We have to delete this config provided by private_message since it was
-  // moved to an optional submodule that will be enabled and this will make
-  // error:
-  // Configuration objects (message.template.private_message_notification)
-  // provided by private_message_notify already exist in active configuration.
-  $config = \Drupal::configFactory()->getEditable('message.template.private_message_notification');
-  if (!$config->isNew()) {
+  // Removed hook due to database update error.
+  // @see https://www.drupal.org/project/social/issues/3266049
+}
+
+/**
+ * Disable the default message template provided by private_message.
+ */
+function social_private_message_update_11007(): void {
+  // Removed hook due to database update error and moved code to
+  // social_private_message_update_11011.
+  // @see https://www.drupal.org/project/social/issues/3266049
+}
+
+/**
+ * Delete the configs that will be migrated to private_message_notify.
+ */
+function social_private_message_update_11008(): void {
+  // Only delete configurations if private_message_notify is not installed
+  // already.
+  if (\Drupal::moduleHandler()->moduleExists('private_message_notify')) {
+    return;
+  }
+
+  // We should delete configurations that will be migrated to
+  // private_message_notify to prevent database update errors.
+  $configs = [
+    'core.entity_form_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.mail_body',
+    'core.entity_view_display.message.private_message_notification.mail_subject',
+    'field.field.message.private_message_notification.field_message_pm_thread',
+    'field.field.message.private_message_notification.field_message_private_message',
+    'field.storage.message.field_message_pm_thread',
+    'field.storage.message.field_message_private_message',
+    'message.template.private_message_notification',
+  ];
+
+  $config_factory = \Drupal::configFactory();
+
+  // Some users could edit the configuration, so we need to save it temporarily
+  // and update after private_message_notify will be enabled.
+  // @see social_private_message_update_11010
+  foreach ($configs as $name) {
+    $config = $config_factory->getEditable($name);
+    if ($config->isNew()) {
+      continue;
+    }
+
+    // Set config data to a new temporary config.
+    $config_tmp = $config_factory->getEditable("{$name}.tmp");
+    $config_tmp->setData($config->getRawData())->save();
+
+    // Delete config.
     $config->delete();
   }
 }
 
+/**
+ * Enable notifications submodule.
+ */
+function social_private_message_update_11009(): void {
```

Comment:
Isn't this the same as what happens within `private_message_update_8007`, so isn't this obsolete if we already put in the dependencies in 

```
// Make sure that we delete/update private_message configurations before
  // private_message_notify will be enabled in the private_message module.
  $dependencies['private_message'][8007] = [
    'social_private_message' => 11008,
  ];
  ```
  
  Or why would there be a scenario that this still is required?

Reply:
Ideally you could say that `function social_private_message_update_11010(): void {`
is depending on the module first being enabled

Reply:
We have two cases:
- Users who installed OS 11.0.3 from scratch and they have no errors (private_message_update_8007 passed)
- Users who tried update from 11.0.2 to 11.0.3 and have errors, so they are still on 11.0.2 (private_message_update_8007 didn't pass)

Since we don't use for now a patch for PM with these changes https://git.drupalcode.org/project/private_message/-/merge_requests/41 we just make sure that our hook updates will execute first and we do all on our side.